kmm/kmm-post-data/src/commonMain/kotlin/com/gchristov/newsfeed/kmmpostdata/usecase/DecoratePostUseCase.kt

```diff
@@ -0,0 +1,74 @@
+package com.gchristov.newsfeed.kmmpostdata.usecase
+
+import com.gchristov.newsfeed.kmmpostdata.Post
+import com.gchristov.newsfeed.kmmpostdata.PostRepository
+import com.gchristov.newsfeed.kmmpostdata.model.DecoratedPost
+import com.gchristov.newsfeed.kmmpostdata.util.ReadingTimeCalculator
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import kotlinx.datetime.Instant
+
+class DecoratePostUseCase(
+    private val postRepository: PostRepository,
+    private val dispatcher: CoroutineDispatcher
+) {
+
+//    suspend fun decoratedPost(
+//        postId: String
+//    ): DecoratedPost {
+//
+//        return postRepository.run {
+//            cachedPost(postId)?.let { post ->
+//                clearCache(postId)
+//                decoratePost(post)
+//            } ?: queryNewPost(postId)
+//        }
+//    }
+
+    suspend fun decoratedPost(
+        postId: String
+    ): DecoratedPost {
+
+        return postRepository.run {
+//            cachedPost(postId)?.let { post ->
+//                clearCache(postId)
+//                decoratePost(post)
+//            } ?:
+            queryNewPost(postId)
+        }
+    }
+
+    suspend fun cachedPost(postId: String): DecoratedPost? =
+        postRepository.cachedPost(postId)?.let { post ->
+            decoratePost(post)
+        }
+
+    suspend fun clearCache(postId: String) = postRepository.clearCache(postId)
+
+    suspend fun redecoratePost(post: DecoratedPost): DecoratedPost = decoratePost(post.raw)
```

Comment:
How about the `onToggleFavourite` shall it `getPost` with the `onCache` callback too there?

Reply:
Not sure I understood the question here 😅

My original point was that since we have a separate use case for decoration now maybe we don’t need a redecorate method at all since we can just call the entire use case again from the view-model when we want to redecorate.

Reply:
This answers the question - in `onToggleFavourite` in `PostViewModel` I'll call `getPost` again (with the `onCache` callback)

Reply:
Ah got you, i think that makes sense.

Reply:
This should've been addressed - just repeated the call to `getPost`